src/omero_zarr/extinfo.py

```diff
@@ -0,0 +1,229 @@
+import re
+
+from omero.gateway import BlitzGateway, BlitzObjectWrapper, ImageWrapper
+from omero.model import Dataset, ExternalInfoI, Image, ImageI, Plate, Project, Screen
+from omero.rtypes import rlong, rstring
+from omero_sys_ParametersI import ParametersI
+
+# Regex to match well positions (eg. A1)
+WELL_POS_RE = re.compile(r"(?P<row>\D+)(?P<col>\d+)")
+
+
+def get_extinfo(conn: BlitzGateway, image: ImageWrapper) -> ExternalInfoI:
+    """
+    Get the external info for an OMERO image.
+
+    Args:
+        conn (BlitzGateway): Active OMERO gateway connection
+        image (ImageWrapper): OMERO image
+
+    Returns:
+        ExternalInfoI: External info object
+
+    Raises:
+        Exception: If the query fails or the path cannot be retrieved
+    """
+
+    details = image.getDetails()
+    if details and details._externalInfo:
+        params = ParametersI()
+        params.addId(details._externalInfo._id)
+        query = """
+            select e from ExternalInfo as e
+            where e.id = :id
+        """
+        extinfo = conn.getQueryService().findByQuery(query, params)
```

Comment:
In think this needs to be a cross-group query `findByQuery(query, params, conn.SERVICE_OPTS)` otherwise you won't see it if the extinfo isn't in your default group.

Reply:
This is still outstanding. Same for the other `conn.getQueryService().findByQuery` below too.

Reply:
Ah ok, then you have to pass `conn.SERVICE_OPTS` that into the function call again. Ok, i'll fix that.